File: DailyLeetcode/AnythingLOL/2064.cpp
class Solution {
public:
  int minimizedMaximum(int n, vector<int> &quantities) {
    int l = 1;
    int r = 0;
    for (int &num : quantities) {
      r = max(r, num);
    }

    int result = -1;
    while (l <= r) {
      int m = l + (r - l) / 2;
      int sum = 0;
      for (int &num : quantities) {
        sum += (int)ceil((float(num) / m));
      }

      if (sum <= n) {
        result = m;
        r = m - 1;
      } else {
        l = m + 1;
      }
    }
    return result;
  }
};

File: src/DailyLeetcode/AnythingLOL/2064.rs
use std::cmp::max;

impl Solution {
    pub fn minimized_maximum(n: i32, quantities: Vec<i32>) -> i32 {
        let mut l = 1;
        let mut r = 0;

        for &num in quantities.iter() {
            r = max(r, num);
        }

        let mut result = -1;
        while l <= r {
            let mut m = l + (r - l) / 2;
            let mut temp = 0;
            for &num in quantities.iter() {
                temp += (num as f64 / m as f64).ceil() as i32;
            }

            if (temp <= n) {
                result = m;
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        result
    }
}
